### PR TITLE
add error message if we add item that already exists in cart

### PR DESCRIPTION
**Pull request by shahad-yousif**

### CODE REVIEW

**Comment by MwMaki** [comment]
Please finish this in-order to not have many conflicts

**Comment by shahad-yousif** [comment]
i am working on

File: app/controllers/api/v1/order_items_controller.rb [hunk 1]
```diff
@@ -2,12 +2,17 @@ class Api::V1::OrderItemsController < ApplicationController
 
   def create
     customer = Customer.find(params[:customer_id])
-    order_item = OrderItem.new(product_id: params[:product_id], cart_id: customer.cart.id, quantity: params[:quantity])
-    authorize order_item
-    if order_item.save
-      render json: {id:order_item.id, status: "#{I18n.translate('activerecord.attributes.controller.messages.add_to_cart')}"}.merge(main_info: main_info), status: :ok
+    if customer.cart.order_items.pluck(:product_id).include?(order_item_params[:product_id].to_i)
```
Comment (ssharba) [inline app/controllers/api/v1/order_items_controller.rb]: see if it still working without to_i

File: app/controllers/api/v1/order_items_controller.rb [hunk 2]
```diff
@@ -2,12 +2,17 @@ class Api::V1::OrderItemsController < ApplicationController
...
+      order_item = customer.cart.order_items.where(product_id: order_item_params[:product_id]).last
+      order_item.update(quantity: order_item.quantity + order_item_params[:quantity].to_i)
```
Comment (ssharba) [inline app/controllers/api/v1/order_items_controller.rb]: here too

**Pull request merged**